hypothesis-python/tests/cover/test_stateful.py

```diff
@@ -1320,7 +1326,106 @@ def rule1(self, data):
 TestLotsOfEntropyPerStepMachine = LotsOfEntropyPerStepMachine.TestCase
 
 
-def test_flatmap():
+def test_filter():
+    class Machine(RuleBasedStateMachine):
+        a = Bundle("a")
+
+        @initialize(target=a)
+        def initialize(self):
+            return multiple(1, 2, 3)
+
+        @rule(
+            a1=a.filter(lambda x: x < 2),
+            a2=a.filter(lambda x: x > 2),
+            a3=a,
+        )
```

Comment:
For clarity, I suggest naming the bundle `val` and the arguments `a, b, c` - it's easier to misread the error logs when every single name matches `a_?\d`!